frontend/ui/widget/imagewidget.lua

```diff
@@ -385,19 +424,23 @@ function ImageWidget:paintTo(bb, x, y)
     end
     if do_alpha then
         -- NOTE: MuPDF feeds us premultiplied alpha (and we don't care w/ GifLib, as alpha is all or nothing),
-        -- but NanoSVG feeds us straight alpha
+        --       while NanoSVG feeds us straight alpha
         if self._is_straight_alpha then
-            --- @todo if Screen.sw_dithering then use bb:ditheralphablitFrom() when it's available
-            bb:alphablitFrom(self._bb, x, y, self._offset_x, self._offset_y, size.w, size.h)
+            -- NOTE: Our icons are already preprocessed properly, either at encoding time, or at caching time.
```

Comment:
so we don't go thru here anymore as you reset `self._is_straight_alpha` ? (might be worth adding to your comment?)

Reply:
That's the "at caching time" ^^.

Reply:
It's just the context where that NOTE is: it's inside `if self._is_straight_alpha then` but talks about our icons... not saying precisely we won't ever go thru this `if` branch. But the presence of the NOTE inside the `if` makes me think it applies to this branch, that our icons will go thur it (might be just me :) but it confused me when reading it)

Reply:
Right, the comment actually applies to "why are we never dithering on icons" bit of the logic.

straight vs. pmul doesn't really enter the equation, as both are handled, which makes the fact that, yes, we won't actually get straight alpha with icons anymore somewhat irrelevant?

Reply:
That's because I tend to comment branches *before* the if when I can.

I can move it inside the branch if that helps?

Reply:
I was expecting it outside :) or I dunno which branch you're talking about :)
I think this comment just needs to be more explicite than just stating a fact: `Our icons are already preprocessed properly`. _Yay, so what? why tell me that here at this point? what should I make out of that? is there a trick?_ (that was my thinking :)
It needs an additional `so, we don't have self._is_straight_alpha on them, and we go / don't go thru here or there`

Reply:
But the two are entirely orthogonal ^^.

The straight alpha thingy is not at all the point of the comment ^^.

The point of the comment is to explain why it skips dithering for icons.

Reply:
```suggestion
            --- @note: Our icons are already preprocessed properly, either at encoding time, or at caching time.
```

Reply:
Actually never mind… ish, it's only fixme, todo and warning that are LDoc. >_>

But elsewhere you've used `@note`.

Reply:
Tweaked the comments a bit ;).

----

Also, that made me realize that we weren't actually skipping alpha-blending on BB32, so I fixed that by flipping the alpha flag in IconWidget ;).

Reply:
> But the two are entirely orthogonal ^^.

I know, they were.

> The straight alpha thingy is not at all the point of the comment, and completely irrelevant: if it were still set, this would still be true and work just as well ^^.

Yes, but the point that you talk about icons in a branch where we never go with icons is confusing :) (and I agree that if we were going thru it, the comment is right).

(As you usually don't mind getting verbose in comments, may be just adding `(although we don't go thru here currently as we set straight_alpha=nil for all our icons :-P)` would make me confident I won't get stuck in the future - but ok anyways :)

Reply:
> Actually never mind… ish, it's only fixme, todo and warning that are LDoc. >_>
> 
> But elsewhere you've used `@note`.

Huh :D. Well, can't hurt anyway ;).

Reply:
> (As you usually don't mind getting verbose in comments, may be just adding `(although we don't go thru here currently as we set straight_alpha=nil for all our icons :-P)` would make me confident I won't get stuck in the future - but ok anyways :)

Sure, that should be part of the latest round of commits ;).